### PR TITLE
v0.1.0 Release

### PR DESCRIPTION
**Pull request by noheltcj**

### CODE REVIEW

**Review by Antim4tter — DISMISSED** [review]

**Review by marniuma — DISMISSED** [review]

File: common/src/main/kotlin/com/noheltcj/rxcommon/subjects/Subject.kt [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+import com.noheltcj.rxcommon.Source
+import com.noheltcj.rxcommon.disposables.CompositeDisposeBag
+import com.noheltcj.rxcommon.disposables.Disposable
+import com.noheltcj.rxcommon.disposables.Disposables
+import com.noheltcj.rxcommon.emitters.Emitter
+import com.noheltcj.rxcommon.observers.AbstractObserver
+import com.noheltcj.rxcommon.observers.Observer
+
+abstract class Subject<E> : AbstractObserver<E>(), Source<E>, Disposable {
```
Comment (noheltcj) [inline common/src/main/kotlin/com/noheltcj/rxcommon/subjects/Subject.kt]: No need to use AbstractObserver here.

File: build.gradle [hunk 2]
```diff
@@ -1,7 +1,7 @@
 buildscript {
     ext {
         project_group_id = 'com.noheltcj'
-        project_version = '0.0.2'
+        project_version = '0.1.0'
         kotlin_version = '1.2.60'
```
Comment (tripleentendre) [inline build.gradle]: Might as well update to `1.2.61`

File: common/src/main/kotlin/com/noheltcj/rxcommon/emitters/ColdEmitter.kt [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    activeObservers.forEach(Observer<E>::onDispose)
+  }
+
+  private fun release() {
+    released = true
+    activeObservers.forEach { observer ->
+      forwardPressure.forEach { element ->
+        observer.onNext(element)
+      }
+      if (isCompleted)
```
Comment (tripleentendre) [inline common/src/main/kotlin/com/noheltcj/rxcommon/emitters/ColdEmitter.kt]: should single line this

File: common/src/main/kotlin/com/noheltcj/rxcommon/emitters/ColdEmitter.kt [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  override fun terminate(throwable: Throwable) {
+    isTerminated = true
+    terminalError = throwable
+    if (released)
+      activeObservers.forEach { it.onError(throwable) }
+  }
+
+  override fun complete() {
+    isCompleted = true
+    if (released)
```
Comment (tripleentendre) [inline common/src/main/kotlin/com/noheltcj/rxcommon/emitters/ColdEmitter.kt]: should single line this

**Pull request merged**